src/geometamaker/cli.py

```diff
@@ -0,0 +1,155 @@
+
+
+def echo_validation_error(error, filepath):
+    summary = u'\u2715' + f' {filepath}: {error.error_count()} validation errors'
+    click.secho(summary, fg='bright_red')
+    for e in error.errors():
+        location = ', '.join(e['loc'])
+        msg_string = (f"    {e['msg']}. [input_value={e['input']}, "
+                      f"input_type={type(e['input']).__name__}]")
+        click.secho(location, bold=True)
+        click.secho(msg_string)
+
+
+@click.command(
+    help='''Validate a .yml metadata document given by FILEPATH.
+    Or if FILEPATH is a directory, validate all documents within.''',
+    short_help='Validate metadata documents for syntax or type errors.')
+@click.argument('filepath', type=click.Path(exists=True))
+@click.option('-r', '--recursive', is_flag=True, default=False,
+              help='if `filepath` is a directory, validate documents '
+                   'in all subdirectories.')
+def validate(filepath, recursive):
+    if os.path.isdir(filepath):
+        file_list, message_list = geometamaker.validate_dir(
+            filepath, recursive=recursive)
+        for filepath, msg in zip(file_list, message_list):
+            if isinstance(msg, ValidationError):
+                echo_validation_error(msg, filepath)
+            else:
+                color = 'yellow'
+                icon = u'\u25CB'
+                if not msg:
+                    color = 'bright_green'
+                    icon = u'\u2713'
+                click.secho(f'{icon} {filepath} {msg}', fg=color)
+    else:
+        error = geometamaker.validate(filepath)
+        if error:
+            echo_validation_error(error, filepath)
+
+
+def print_config(ctx, param, value):
+    if not value or ctx.resilient_parsing:
+        return
+    config = geometamaker.Config()
+    click.echo(config)
+    ctx.exit()
+
+
+def delete_config(ctx, param, value):
+    if not value or ctx.resilient_parsing:
+        return
+    config = geometamaker.Config()
+    click.confirm(
+        f'Are you sure you want to delete {config.config_path}?',
+        abort=True)
+    config.delete()
+    ctx.exit()
+
+
+@click.command(
+    short_help='''Configure GeoMetaMaker with information to apply to all
+    metadata descriptions''',
+    help='''When prompted, enter contact and data-license information
+    that will be stored in a user profile. This information will automatically
+    populate contact and license sections of any metadata described on your
+    system. Press enter to leave any field blank.''')
+@click.option('--individual-name', prompt=True, default='')
+@click.option('--email', prompt=True, default='')
+@click.option('--organization', prompt=True, default='')
+@click.option('--position-name', prompt=True, default='')
+@click.option('--license-title', prompt=True, default='',
+              help='the name of a data license, e.g. "CC-BY-4.0"')
+@click.option('--license-path', prompt=True, default='',
+              help='a url for a data license')
```

Comment:
The parameter name `path` and the use of `url` in the help text make me wonder about how the text should be provided.  Should it be a URL?   A path to a local file?  A URI instead?  Maybe an example or two would help clarify?

Reply:
yeah good point. I think a URL makes the most sense. `path` here comes from the DataPackage schema (https://specs.frictionlessdata.io/data-package/#licenses) We're sort of trying to mirror that where possible, for ease of future interoperability. But I think it makes sense to change the user-facing name to `--license-url` in our case.